include/math_utils.h

```diff
@@ -252,14 +252,14 @@ constexpr uint32_t clamp_to_uint32(const T val)
 	return static_cast<uint32_t>(std::clamp(val, min_val, max_val));
 }
 
-constexpr uint8_t read_low_nibble(const uint8_t byte)
+constexpr int low_nibble(const int byte)
 {
-	return static_cast<uint8_t>(byte & 0x0f);
+	return byte & 0x0f;
 }
 
-constexpr uint8_t read_high_nibble(const uint8_t byte)
+constexpr int high_nibble(const int byte)
 {
-	return static_cast<uint8_t>(byte >> 4);
+	return (byte & 0xf0) >> 4;
 }
```

Comment:
I get your reasoning on moving stuff to int but I don't like it for functions doing bitwise operations.  As an example, you can run into problems because C will do an arithmetic right shift on signed integers when you usually will want a logical right shift.  It works out to the same thing here but you had to put in a bitwise AND to counteract that.

Also, I don't think the `static_cast` was required in the original code.

Reply:
I was on the fence with this one. I'm happy to restore it to accept `uint8_t`s only, I just thought this might more convenient. But probably we'll always dealing with `uint8_t` at the call site anyway when messing around with nibbles.

Technically, I don't think the signed bit would cause problems as I'm doing the AND before the shift, but moot point 😄